app/src/apdu_handler.c

```diff
@@ -33,11 +33,19 @@
 #include "parser_impl.h"
 
 void extractHDPath(uint32_t rx, uint32_t offset) {
-    if ((rx - offset) < sizeof(uint32_t) * HDPATH_LEN_DEFAULT) {
+    if ((rx - offset) == sizeof(uint32_t) * HDPATH_LEN_3) {
+        hdPathLen = HDPATH_LEN_3;
+    } else if ((rx - offset) == sizeof(uint32_t) * HDPATH_LEN_DEFAULT) {
+        hdPathLen = HDPATH_LEN_DEFAULT;
+    } else {
         THROW(APDU_CODE_WRONG_LENGTH);
     }
 
-    MEMCPY(hdPath, G_io_apdu_buffer + offset, sizeof(uint32_t) * HDPATH_LEN_DEFAULT);
+    char buffer[15];
+    snprintf(buffer, sizeof(buffer), "Path len: %d", hdPathLen);
+    zemu_log_stack(buffer);
+
+    MEMCPY(hdPath, G_io_apdu_buffer + offset, sizeof(uint32_t) * hdPathLen);
 
     const bool mainnet = hdPath[0] == HDPATH_0_DEFAULT &&
```

Comment:
in case the path has three values (HDPATH_LEN_3 / ADR8), we must check that all values are hardened ( >= 0x8000'0000)

Reply:
in this particular case given we only accept 
- hdPath[0] = HDPATH_0_DEFAULT
- hdPath[1] = HDPATH_1_DEFAULT
please check that `hdPath[2] >= 0x80000000` when `hdPathLen==3`

Reply:
Done!

Reply:
I am not sure where this is 😂

Reply:
It is [here](https://github.com/Zondax/ledger-oasis/blob/ddf4bcc1029c38f47c4ee0c7b5cb9398a1eb5aa7/app/src/apdu_handler.c#L57-L59)

Reply:
silly me :)